subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformArtifactOperation.java

```diff
@@ -29,31 +30,35 @@
 
 class TransformArtifactOperation implements RunnableBuildOperation {
     private static final Logger LOGGER = LoggerFactory.getLogger(TransformArtifactOperation.class);
-    private final ResolvableArtifact artifact;
+    private final ComponentArtifactIdentifier artifactId;
+    private final File file;
     private final ArtifactTransformer transform;
     private Throwable failure;
     private List<File> result;
 
-    TransformArtifactOperation(ResolvableArtifact artifact, ArtifactTransformer transform) {
-        this.artifact = artifact;
+    TransformArtifactOperation(ComponentArtifactIdentifier artifactId,  File file, ArtifactTransformer transform) {
+        this.artifactId = artifactId;
+        this.file = file;
         this.transform = transform;
     }
 
     @Override
     public void run(@Nullable BuildOperationContext context) {
```

Comment:
`context` should be `Nullable` in `TransformFileOperation`, too.

Reply:
Fixed.